### ISSUE TITLE
false positive with MA0109

### ISSUE THREAD

**Issue by flucas1** [meziantou/Meziantou.Analyzer#1253]
### Version of the Meziantou.Analyzer NuGet package

3.0.136

### Rule Identifier

MA0109

### Target Framework

.net11

### C# Language version

C#15 preview

### .NET SDK version

.net 11 preview 6

### Description

This code raises MA0109

```
        public static async Task<bool> TestSpanParameterAsync(string[] pathElements)
        {
            await Task.CompletedTask.ConfigureAwait(false);
            return false;
        }
```

Ok, so by adding a span overload such as

```
        public static async Task<bool> TestSpanParameterAsync(Span<string> pathElements)
        {
            await Task.CompletedTask.ConfigureAwait(false);
            return false;
        }
```

I get "CS4012: Parameters of type cannot be declared in async methods or async lambda expressions." on the overload.

I understand that MA0109 should exclude async methods.

### Reproduction Steps

.

**Comment by flucas1** [comment]
`Memory<T>` is OK
